build_tools/benchmarks/run_benchmarks_on_android.py

```diff
@@ -272,11 +277,15 @@ def run_benchmarks_for_category(
     benchmark_category_dir: str,
     benchmark_case_dirs: Sequence[str],
     normal_benchmark_tool: str,
-    traced_benchmark_tool: Optional[str],
-    trace_capture_tool: Optional[str],
-    verbose: bool = False
-) -> Sequence[Tuple[BenchmarkInfo, Dict[str, Any], Dict[str, Any],
-                    Optional[str]]]:
+    tmp_dir: str,
```

Comment:
This function makes me a bit sad now.. We need to see how to break it down. (Not requesting for doing that right now.)

Reply:
Worth having a doc for `tmp_dir` in the docstring at least. (The other new ones are relatively self-explanatory.)

Reply:
> LGTM, thanks for adding this! But given this keep going feature does not have some count down, I assume it's only meant for local usage.

Not sure what you mean by a countdown? My main use case was for running locally, although I think it would also be nice to be able to do this on CI so you could see the results. It would require plumbing more things through though, so I wouldn't do it right now.

> My main comment is mostly around having more comments explaining the intent. I know it's an art as when/where to comment to avoid being excessive; but generally if the logic is too branchy and lengthy I'd suggest to put more to keep following the logic. :)

Yeah, like I said, I don't really love this. There are a lot of switches and parameters all over the place :confused: I'll add some more comments and docstrings.